### PR TITLE
Feature: Clearer Rules & Prettier Hamburger Menu

### PR DESCRIPTION
**Pull request by CarmenUZH**
I changed a lot of style components, sorry

### CODE REVIEW

File: src/components/ui/ColorButtons.scss [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.colorbuttons{
```
Comment (BacLuc) [inline src/components/ui/ColorButtons.scss:1]:
```suggestion
.colorbuttons {
```
I would keep the formatting and put a space here.
(and separate the words with dashes)

File: src/components/ui/ColorButtons.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  position: fixed;
```
Comment (BacLuc) [inline src/components/ui/ColorButtons.scss:2]: position fixed for all buttons?
Why?

File: src/components/ui/ColorButtons.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  transition: all .4s ease-in-out;
+  &:hover{
+    background-position: 100% 0;
+    -moz-transition: all .4s ease-in-out;
+    -o-transition: all .4s ease-in-out;
+    -webkit-transition: all .4s ease-in-out;
+    transition: all .4s ease-in-out;
+  }
+}
+.bucolor1{
```
Comment (BacLuc) [inline src/components/ui/ColorButtons.scss:26]: and which color is this?

File: src/components/ui/ColorButtons.scss (hunk 1 above)
Comment (BacLuc) [inline src/components/ui/ColorButtons.scss:1]: If we follow the global css pattern (which i don't like), this file should be in src/styles/ui.
I would prefer the component based approach with a component ColorButton which has a parameter vairant (which controls the color), and RulesButton uses the ColorButton and chooses one of the variants.

File: src/components/views/game/room/RoomInfoContainer.tsx [hunk 4]
```diff
@@ -4,32 +4,50 @@ import { useCurrentGame } from "../../../../hooks/api/useCurrentGame"
 import Button from "@mui/material/Button"
```
Comment (BacLuc) [inline src/components/views/game/room/RoomInfoContainer.tsx:4]: With the sopra resolution (1366x768), this does not look good:
grafik

File: src/components/ui/RulesButton.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      <Dialog
+        open={showRules}
+        onClose={clickRules}
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="alert-dialog-title">THESE ARE THE RULES</DialogTitle>
+        <DialogContent>
+          <DialogContentText id="alert-dialog-description">
+            {theRules}
```
Comment (BacLuc) [inline src/components/ui/RulesButton.tsx:52]: react complains that the generated DOM is not valid:

Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.

File: src/components/ui/HamburgerMenu.tsx [hunk 6]
```diff
@@ -1,43 +1,11 @@
...
-import ListAltIcon from "@mui/icons-material/ListAlt"
-import { useCurrentGame } from "../../hooks/api/useCurrentGame"
-import { Paths } from "../routing/routers/Paths"
-import { useNavigate } from "react-router-dom"
-import { Grow, Tooltip } from "@mui/material"
+import { Tooltip, Slide } from "@mui/material"
+import RulesButton from "./RulesButton"
+import LeaveButton from "./LeaveButton"
 
 const HamburgerMenu = () => {
```
Comment (BacLuc) [inline src/components/ui/HamburgerMenu.tsx:7]: the hamburger menu looks nice

File: src/components/ui/HamburgerMenu.tsx [hunk 7]
```diff
@@ -53,31 +21,30 @@ const HamburgerMenu = () => {
...
+            style={{
+              marginTop: "10%",
+              color: "white",
+              left: "5%",
+              width: "80%",
+            }}
+          />
+        </ul>
+      </Slide>
+      <p className={"hamburgerinfotext"}>
```
Comment (BacLuc) [inline src/components/ui/HamburgerMenu.tsx:44]: This overlaps with other texts.

grafik

Maybe you could use a tooltip for that, where you don't use the hover trigger, but control it yourself?

File: src/components/ui/HamburgerMenu.scss [hunk 8]
```diff
@@ -1,61 +1,47 @@
 
-
-.navBar {
-  position: relative;
-  overflow: hidden;
+.hamburgerinfotext{
```
Comment (BacLuc) [inline src/components/ui/HamburgerMenu.scss:2]:
```suggestion
.hamburgerinfotext {
```

**Review by BacLuc — CHANGES_REQUESTED** [review]
Major:
- fix issues with the 1366x768 resolution
- fix the overlapping info text for the menu
- generate valid dom when displaying the rules
- is position fixed for the buttons really needed?

Minor:
- naming (dashes vs allwordsgluedtogetherandnotreadable) and architectural consideration
- css formatting

File: src/components/ui/HamburgerMenu.tsx (hunk 7 above)
Comment (CarmenUZH) [reply src/components/ui/HamburgerMenu.tsx:44]: I fixed it by creating a second Tool tip that disappears after 5 seconds - im not sure how good the quality of this code is so please review the changes in HamburgerMenu.tsx when i push this next commit

File: src/components/views/game/room/RoomInfoContainer.tsx (hunk 4 above)
Comment (CarmenUZH) [reply src/components/views/game/room/RoomInfoContainer.tsx:4]: Fixed this! It looks a bit smaller on the "normal" resolution now, but at least the SoPra resolution looks presentable now 😂

File: src/components/views/game/room/Room.scss [hunk 9]
```diff
@@ -1,5 +1,14 @@
 @import 'src/styles/theme';
 
+
 .room {
   padding: 0 20px;
 }
+
+.roominfobackground{
```
[the remark this answers is not in the document]
Comment (CarmenUZH) [reply src/components/views/game/room/Room.scss:8]: okay! changing this wherever i saw it

File: src/components/ui/ColorButtons.scss (hunk 1 above)
Comment (CarmenUZH) [reply src/components/ui/ColorButtons.scss:1]: refactored this!

File: src/components/ui/ColorButtons.scss (hunk 2 above)
Comment (CarmenUZH) [reply src/components/ui/ColorButtons.scss:2]: probably something i forgot to remove when i copied some code around, deleted it

File: src/components/ui/ColorButtons.scss (hunk 3 above)
Comment (CarmenUZH) [reply src/components/ui/ColorButtons.scss:26]: added clearer names

File: src/components/ui/ColorButtons.scss (hunk 1 above)
Comment (CarmenUZH) [reply src/components/ui/ColorButtons.scss:1]: We can talk about this during our next meeting

File: src/components/ui/RulesButton.tsx (hunk 5 above)
Comment (CarmenUZH) [reply src/components/ui/RulesButton.tsx:52]: I changed the <p> tag from "theRules" into an empty tag - You don't get the error message anymore and everything still looks the same, but please tell me if this is somehow a bad solution

File: src/components/ui/HamburgerMenu.scss (hunk 8 above)
Comment (CarmenUZH) [reply src/components/ui/HamburgerMenu.scss:2]: Got deleted because of the previous feedback

**Comment by elBeato** [comment]
Grundsätzlich gut, ich würden den Farbverlauf weglassen. Aber von mir aus können wir das auch später noch anpassen.

File: src/components/ui/HamburgerMenu.scss [hunk 10]
```diff
@@ -67,23 +15,19 @@
...
   max-width: 290px;
   z-index: 9;
+  align-items: center;
 }
 
 .menuNav.showMenu {
   height: 110%;
   margin-top: -10px;
   padding-left: 40px;
   background: linear-gradient(90deg, rgba(156, 20, 10, 1) 0% ,rgba(150, 25, 15, 0.6) 100% );
```
Comment (elBeato) [inline src/components/ui/HamburgerMenu.scss:25]: Würde ich einfarbig machen.

**Review by elBeato — APPROVED** [review]

**Pull request merged**